File: src/11. Container With Most Water/soln.java
class Solution {
    public int maxArea(int[] height) {
        int ans = 0;
        int l = 0, r = height.length - 1;

        while (l < r) {
            int minHeight = Math.min(height[l], height[r]);
            ans = Math.max(ans, minHeight * (r - l));

            if (height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }

        return ans;
    }
}

File: src/11. Container With Most Water/soln.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans = 0
        l, r = 0, len(height) - 1

        while l < r:
            min_height = min(height[l], height[r])
            ans = max(ans, min_height * (r - l))

            if height[l] < height[r]:
                l += 1
            else:
                r -= 1

        return ans

        
